### ISSUE TITLE
Empty getMapObjects response

### ISSUE THREAD

**Issue by kilabyte** [lsapan/pgoapi-swift#23]
Well looks like Niantic finally change their API today. Looks like in the response object we still get mapCells but now they have it in a nested tuple.

AHAAAAAAA/PokemonGo-Map/issues/3403@237372013

NECROBOTIO/NecroBot@35339c3

**Comment by spinach** [comment]
Can follow status here: 
https://www.reddit.com/r/pokemongodev/comments/4w1cvr/pokemongo_current_api_status/

Latest update is:
`It's becoming more evident that this is a non-trivial change, and will take much longer than planned to get reverse engineered again.`

**Comment by PokemonGoSucks** [comment]
I have a feeling it is trivial. Might just be the request envelope.

**Comment by lsapan** [comment]
I have a feeling POGOProtos will be updated by the time we all wake up tomorrow haha. I'll keep an eye on the repo and update as changes are made.

**Comment by lsapan** [comment]
Just to update: never take what you hear from skimming as truth. The protos and format have not changed. The tuple thing you found was a change that a different project had made internally, totally unrelated.

What has changed is one of the values that gets sent along. The POGOProtos team seems to have limited it down to `Unknown6`, but they're still actively discussing / working through it.

For those who want updates, follow this issue:
https://github.com/AeonLucid/POGOProtos/issues/131

**Comment by dmallory** [comment]
Fixed API: https://github.com/keyphact/pgoapi

**Comment by PokemonGoSucks** [comment]
Yes, we will port it.. (Will start in an hour)

**Comment by tomalbrc** [comment]
Isn't there a problem with the encrypt.c containing niantic's copyrighted code? How far have you come with porting?

**Comment by PokemonGoSucks** [comment]
It will be a swift interpretation for software compatibility, which is fine in my jurisdiction. However, using the API is another question and I'm not responsible for what you do with it 😉. I am still porting and have a lot left... I have xxhash ported to swift so far and I'm onto the pseudo-c signature library (which I'm surprised they kept in its form). I have work all day, but might PR later tonight/tomorrow. If anyone else wants to build on what I have, I'll update my fork. There are plenty of other changes to make, beyond unknown6, to emulate the app more closely as well..

**Comment by lsapan** [comment]
@PokemonGoSucks thanks for taking this on. Feel free to submit a PR that's even part way there and we can assist.

**Comment by bmoliveira** [comment]
I've started to migrate the changes to support the new version support, but I'm not being able to port the xxhash from the c libs I've found does anyone found a way to port it and managed it to work?

**Comment by kilabyte** [comment]
Check out https://github.com/AgentFeeble/pgoapi Looks like he got the hash to work

**Comment by PokemonGoSucks** [comment]
Sorry guys. I am done the implementation, but the c encrypt file has a lot of "too complex to be solved in reasonable time" errors in Swift. I've been busy lately and can't figure out a work around besides breaking up all the 100s of errors, which seems like more trouble than it's worth. How do we feel about using the regular c library?

EDIT: Nevermind, I have a solution..

**Comment by tomalbrc** [comment]
How are you doing? As @lsapan mentioned, we can/want to assist ;)

**Comment by PokemonGoSucks** [comment]
I messed up the encryption somehow after migrating to c code.. :/ Trying to solve it. My fork is being updated.

I just want to get the map objects working again before moving on to porting the code to Swift again and making it compile.

Edit: Turned out to just be a missing variable. If anyone wants a working fork with map objects, see my fork, but it uses the c library. Will not PR until I'm done porting to swift, which I hope to do by tonight.
